code/part-one/mining.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const { createHash } = require('crypto');
+const signing = require('./signing');
+const { Block, Blockchain } = require('./blockchain');
+
+
+class MineableTransaction {
+  constructor(privateKey, recipient = null, amount) {
+    const publicKey = signing.getPublicKey(privateKey);
+    this.amount = amount;
+
+    if (recipient !== null) {
+      this.source = publicKey;
+      this.recipient = recipient;
+    } else {
+      this.source = null;
+      this.recipient = publicKey;
+    }
+
+    const toSign = this.source + this.recipient + amount;
+    this.signature = signing.sign(privateKey, toSign);
+  }
+}
+
+class MineableBlock extends Block {
+  /*
+    Initialize the block constructor
+
+    **Note**
+    The block should be initiated with
+      - a timestamp
+      - transactions
+      - the previous block's hash
+  */
+  constructor(transactions, previousHash) {
+    super(transactions, previousHash);
+    this.hash = '';
+    this.nonce = null;
```

Comment:
I haven't checked one of the later commits that might address this, but it might be worthwhile to include comments about why these two properties in particular are being set to "nothing" here.